src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -94,8 +94,13 @@ const _calculateDistancesAlongRoute = (waypointModelList, airport) => {
 
     for (let i = 0; i < waypointModelList.length; i++) {
         const waypoint = waypointModelList[i];
-        const waypointPosition = waypoint.relativePosition;
         let previousWaypoint = waypoint;
+
+        if (waypoint.isVector || previousWaypoint.isVector) {
```

Comment:
Yeah, that does seem fishy. Cleaned up the method a bit (though it could still use some attention. Added a `TODO` for the use of `continue`).